src/lib.rs

```diff
@@ -126,11 +46,20 @@ impl Collector {
     ///
     /// ```
     /// # use metrics_process::Collector;
-    /// let collector = Collector::default().prefix("my_prefix_");
+    /// let collector = Collector::new("my_prefix_");
     /// ```
-    pub fn prefix(mut self, prefix: impl Into<String>) -> Self {
-        self.prefix = prefix.into();
-        self
+    pub fn new(prefix: impl AsRef<str>) -> Self {
+        let prefix = prefix.as_ref();
+        Self {
+            cpu_seconds_total: format!("{prefix}process_cpu_seconds_total").leak(),
```

Comment:
I don't think it's good idea to use `leak()` here.

Reply:
This way we are allocating the `String` only once, and then reuse it in `gauge!()` macros for metrics names. The `&'static str` also helps -- it unties the lifetime from the `&self` reference.

The only downside of `leak`ing is that the data of `String`s allocated won't be freed until the program exits. In a contrived scenario, where one creates a ton of `Collectors` with `prefix` provided and then drops them this would cause problems. But from my understanding, the `Collector` is meant to be created once at the start and used for the entire runtime of the program.

Still, we can counter this by `Box::leak`ing the strings allocated and then giving the `Collector` the `Drop` `impl` that will `unsafe`ly `Box::from_raw` them to drop properly. If you're fine with that, I'll implement this in differend PR.

Reply:
How about utilizing `String` to store each label instead of `&'static str` in `Collectors`? I'm asking because the advantages of using `&'static str` in `Collectors` are not clear to me. Is there a significant difference in memory usage or any other factors to consider?